katalog/harbor/database/sts.yml

```diff
@@ -23,16 +23,38 @@ spec:
         app: harbor
         component: database
     spec:
+      securityContext:
+        runAsUser: 999
+        fsGroup: 999
+      automountServiceAccountToken: false
+      terminationGracePeriodSeconds: 120
       initContainers:
-        - name: change-permission-of-directory
-          image: busybox
-          imagePullPolicy: Always
+        # as we change the data directory to a sub folder to support psp, the init container here
+        # is used to migrate the existing data. See https://github.com/goharbor/harbor-helm/issues/756
+        # for more detail.
+        # we may remove it after several releases
+        - name: "data-migrator"
+          image: goharbor/harbor-db
+          imagePullPolicy: IfNotPresent
+          command: ["/bin/sh"]
+          args: ["-c", "[ -e /var/lib/postgresql/data/postgresql.conf ] && [ ! -d /var/lib/postgresql/data/pgdata ] && mkdir -m 0700 /var/lib/postgresql/data/pgdata && mv /var/lib/postgresql/data/* /var/lib/postgresql/data/pgdata/ || true"]
+          volumeMounts:
+            - name: database
+              mountPath: /var/lib/postgresql/data
+              subPath: 
+        # with "fsGroup" set, each time a volume is mounted, Kubernetes must recursively chown() and chmod() all the files and directories inside the volume
+        # this causes the postgresql reports the "data directory /var/lib/postgresql/data/pgdata has group or world access" issue when using some CSIs e.g. Ceph
+        # use this init container to correct the permission
+        # as "fsGroup" applied before the init container running, the container has enough permission to execute the command
+        - name: "data-permissions-ensurer"
+          image: goharbor/harbor-db
+          imagePullPolicy: IfNotPresent
           command: ["/bin/sh"]
-          args: ["-c", "chown -R 999:999 /var/lib/postgresql/data"]
+          args: ["-c", "chmod -R 700 /var/lib/postgresql/data/pgdata || true"]
```

Comment:
is the `|| true` from upstream? If the chmod fails I guess the container will go in crashloop, right? is this the wanted behaviour?

Reply:
I double checked the official chart helm and it contains the `|| true` parameter. From the checks carried out on the comment present on the StatefulSet, this command has been configured in this way in order to correctly set the permissions.